src/Symfony/Component/Scheduler/Messenger/SchedulerTransport.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+use Symfony\Component\Scheduler\Exception\LogicException;
+use Symfony\Component\Scheduler\Generator\MessageGeneratorInterface;
+
+/**
+ * @experimental
+ */
+class SchedulerTransport implements TransportInterface, MessageCountAwareInterface
+{
+    public function __construct(
+        private readonly MessageGeneratorInterface $messageGenerator,
+    ) {
+    }
+
+    public function getMessageCount(): int
```

Comment:
The `MessageCountAwareInterface` need for "fail" queues only. It's not possible to send messages in schedule. Why have you decided to impltment it?

Reply:
It was for `messenger:stats`, but not really useful, I will remove.